tools/toolchains/gcc.py

```diff
@@ -80,6 +82,9 @@ def __init__(self, target,  notify=None, macros=None, build_profile=None,
         elif target.core == "Cortex-M7FD":
             self.cpu.append("-mfpu=fpv5-d16")
             self.cpu.append("-mfloat-abi=softfp")
+        elif target.core.startswith("Cortex-M33F"):
+            self.cpu.append("-mfpu=fpv5-sp-d16")
+            self.cpu.append("-mfloat-abi=hard")
```

Comment:
All other floating point units use `softfp`. Please change this to be consistent.

Reply:
Thank you. I have updated the PR to change to softfp